src/main/java/Mikey.java

```diff
@@ -0,0 +1,295 @@
+                    try {
+                        newTodo(taskDescription, Integer.parseInt(taskCompletion));
+                    } catch (StringIndexOutOfBoundsException e) {
+                        System.out.println("Oi mate, I can't create an empty task yea?");
+                        System.out.println();
+                    }
+                } else if (taskType.equalsIgnoreCase("D")) {
+                    try {
+                        String[] taskDateArray = Arrays.copyOfRange(taskDetails, 7, taskDetails.length);
+                        StringBuffer taskDateJoiner = new StringBuffer();
+                        for (int i = 0; i < taskDateArray.length; i++) {
+                            taskDateJoiner.append(taskDateArray[i]);
+                        }
+                        String taskDate = taskDateJoiner.toString();
+                        taskDate = taskDate.replace(")", "");
+                        newDeadline(taskDescription, taskDate, Integer.parseInt(taskCompletion));
+                    } catch (StringIndexOutOfBoundsException e) {
+                        System.out.println("Oi mate, I can't create an empty task yea?");
+                        System.out.println();
+                    }
+                } else if (taskType.equalsIgnoreCase("E")) {
+                    try {
+                        String[] taskDateArray = Arrays.copyOfRange(taskDetails, 6, taskDetails.length);
+                        StringBuffer taskDateJoiner = new StringBuffer();
+                        for (int i = 0; i < taskDateArray.length; i++) {
+                            taskDateJoiner.append(taskDateArray[i]);
+                            taskDateJoiner.append(" ");
+                        }
+                        String taskDate = taskDateJoiner.toString();
+                        taskDate = taskDate.replace(")", "");
+                        newEvent(taskDescription, taskDate, Integer.parseInt(taskCompletion));
+                    } catch (StringIndexOutOfBoundsException e) {
+                        System.out.println("Oi mate, I can't create an empty task yea?");
+                        System.out.println();
+                    }
+                }
+            }
+        } catch (FileNotFoundException e) {
+            fileNotFound("");
+        }
+
+    }
+
+    public static void fileNotFound(String taskFormatter) {
+        System.out.println("Ooh deary me bruv, I reckon that file don't exists yet, I'm gonna go ahead and make one");
+        try {
+            File newFile = new File("./data/Mikey.txt");
+            newFile.getParentFile().mkdirs();
+            FileWriter listWrite = new FileWriter("./data/Mikey.txt");
+            listWrite.write(taskFormatter.toString());
+            listWrite.close();
+        } catch (IOException e) {
+            System.out.println("Oopsie mate, I can't create a new file");
+        }
+    }
+
+    public static void main(String[] args) throws java.io.IOException {
```

Comment:
Again this function is very long too, so maybe try and refactor some things inside it;